File: scripts/build_site.mjs
import { mkdir, copyFile, cp, writeFile } from 'node:fs/promises';
const root = new URL('../', import.meta.url);
const output = new URL('dist/', root);
await mkdir(output, { recursive: true });
await copyFile(new URL('index.html', root), new URL('index.html', output));
for (const directory of ['src', 'assets']) {
  await cp(new URL(directory, root), new URL(directory, output), { recursive: true });
}
await writeFile(new URL('.nojekyll', output), '');
console.log('Static site built in dist/');

File: scripts/build_site.py
"""Build a minimal static site; never ship the tracked Python environment."""
from pathlib import Path
import shutil
root = Path(__file__).resolve().parents[1]
output = root / 'dist'
output.mkdir(exist_ok=True)
shutil.copy2(root / 'index.html', output / 'index.html')
for directory in ('src', 'assets'):
    shutil.copytree(root / directory, output / directory, dirs_exist_ok=True)
(output / '.nojekyll').write_text('')
print(f'Static site: {output}')
